app/models/manageiq/providers/ibm_cloud/power_virtual_servers/manager_mixin.rb

```diff
@@ -11,16 +11,22 @@ def supported_auth_attributes
 
   def connect(options = {})
     auth_key = authentication_key(options[:auth_type])
-    creds = self.class.raw_connect(auth_key, uid_ems)
-
-    options[:service] ||= "PowerIaas"
-    case options[:service]
-    when "PowerIaas"
-      region, guid, token, crn, tenant = creds.values_at(:region, :guid, :token, :crn, :tenant)
-      IBM::Cloud::SDK::PowerIaas.new(region, guid, token, crn, tenant)
-    else
-      raise ArgumentError, "Unknown target API set: '#{options[:service]}''"
-    end
+    token, power_iaas_service = self.class.raw_connect(auth_key, uid_ems)
+
+    require "ibm_cloud_power"
+
+    # TODO
+    region = "us-south"
```

Comment:
Same comment about the [IBM Cloud Resource Controller API](https://cloud.ibm.com/apidocs/resource-controller/resource-controller).

Reply:
I think this actually is missing in the openapi spec as a property of the service instance, I see region_id in the response payload but it isn't making it to the deserialized object.  Just doing this until I figure out or fix-up the spec and regen the client

Reply:
The region is embedded (as 'location') in the [CRN](https://cloud.ibm.com/docs/account?topic=account-crn), which is included in the openapi spec. We'd have to parse it out.

Reply:
:+1: Parsing the CRN to get the tenant_id and the region, both of those TODOs dropped now